### PR TITLE
Enhancement 714: Expose library methods to list symbols with staged data, and to delete staged data

### LINKED ISSUE #714: Expose Library method to clean up staged data.
**Is your feature request related to a problem? Please describe.**

I'd like to be able to clean up staged writes instead of finalizing them.

```python
df = pd.DataFrame({'column': [5,6,7]}, index=pd.date_range(start='1/1/2000', periods=3))
lib.write('staged', df, staged=True)
# I don't want to finalize, how do I clean up the staged data?
```

If I don't clean up staged data then:
- I have data that I can't see or remove
- I can't prepare any new staged data without it being mixed into the previous data on finalization

**Describe the solution you'd like**

The method exists but isn't part of the public Library API.
```
library._nvs.version_store.remove_incomplete(<symbol>)
```
Decide on naming and expose.

### PR DESCRIPTION
**Pull request by alexowens90**
#### Reference Issues/PRs

Closes #714 

#### What does this implement/fix? How does it work (high level)? Highlight notable design decisions.

Exposes methods on both `NativeVersionStore` and `Library` to list symbols that have staged data, and to delete staged data.

#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [x] Have you updated the relevant docstrings and documentation?
 - [x] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [x] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [x] Are API changes highlighted in the PR description?
 - [x] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

**Review by willdealtry — APPROVED** [review]

File: python/arcticdb/version_store/library.py [hunk 1]
```diff
@@ -810,6 +810,38 @@ def update(
             prune_previous_version=prune_previous_versions,
         )
 
+    def list_symbols_with_staged_data(self) -> List[str]:
+        """
+        Removes staged data.
```
Comment (poodlewars) [inline python/arcticdb/version_store/library.py:815]: Comment is incorrect

File: python/tests/unit/arcticdb/version_store/test_parallel.py [hunk 2]
```diff
@@ -18,11 +18,41 @@
     random_floats,
     random_dates,
 )
+from arcticdb_ext.storage import KeyType
+
+
+def test_remove_incomplete(lmdb_version_store):
```
Comment (poodlewars) [inline python/tests/unit/arcticdb/version_store/test_parallel.py:24]: What happens if you remove incomplete data for a symbol that doesn't exist, or that doesn't have any incomplete data?

File: python/arcticdb/version_store/library.py [hunk 3]
```diff
@@ -810,6 +810,38 @@ def update(
             prune_previous_version=prune_previous_versions,
         )
 
+    def list_symbols_with_staged_data(self) -> List[str]:
```
Comment (jamesmunro) [inline python/arcticdb/version_store/library.py:813]: What's different about this and the other method?

https://github.com/man-group/ArcticDB/blob/master/python/arcticdb/version_store/library.py#L870

Comment (jamesmunro) [reply python/arcticdb/version_store/library.py:813]: My comment about 'seeing staged data' in #714, was not about listing symbols, but rather getting the actual inflight appends and updates.  I think that's lower priority.  At least there is a way to cleanup now.

File: python/arcticdb/version_store/library.py (hunk 1 above)
Comment (alexowens90) [reply python/arcticdb/version_store/library.py]: Function removed as James noticed this functionality already exists:
https://github.com/man-group/ArcticDB/pull/778#discussion_r1300340199

File: python/tests/unit/arcticdb/version_store/test_parallel.py (hunk 2 above)
Comment (alexowens90) [reply python/tests/unit/arcticdb/version_store/test_parallel.py:24]: Added to the test covering these cases (the answer is nothing to both)

File: python/arcticdb/version_store/library.py (hunk 3 above)
Comment (alexowens90) [reply python/arcticdb/version_store/library.py]: No difference, I didn't know it existed. I've removed this method now.

**Review by jamesmunro — APPROVED** [review]

**Pull request merged**